Styra/Opa/OpaClient.cs

```diff
@@ -49,118 +49,138 @@ public OpaClient(string serverUrl)
     /// <returns>Result, as a boolean, or nil in the case of a query failure.</returns>
-    public bool? check(double input, string path)
+    public async Task<bool> check(string path, string input)
     {
-        return query<bool>(input, path);
+        return await evaluate<bool>(path, input);
     }
 
     /// <summary>
     /// Simple allow/deny-style check against a rule, using the provided input list.
     /// </summary>
-    /// <param name="input">The input list OPA will use for evaluating the rule.</param>
+    /// <param name="input">The input List value OPA will use for evaluating the rule.</param>
     /// <param name="path">The rule to evaluate. (Example: "app/rbac")</param>
     /// <returns>Result, as a boolean, or nil in the case of a query failure.</returns>
     /// <remarks>The closest idiomatic type mapping to a JSON Array type for .NET is a List, so we use that here.</remarks>
-    public bool? check(List<object> input, string path)
+    public async Task<bool> check(string path, List<object> input)
     {
-        return query<bool>(input, path);
+        return await evaluate<bool>(path, input);
     }
 
     /// <summary>
-    /// Evaluate a rule, using the provided input map, then coerce the result to type T.
+    /// Simple allow/deny-style check against a rule, using the provided input mapping.
     /// </summary>
-    /// <param name="input">The input Dictionary OPA will use for evaluating the rule.</param>
+    /// <param name="input">The input Dictionary value OPA will use for evaluating the rule.</param>
     /// <param name="path">The rule to evaluate. (Example: "app/rbac")</param>
     /// <returns>Result, as a boolean, or nil in the case of a query failure.</returns>
-    public T? query<T>(Dictionary<string, object> input, string path)
+    /// <remarks>The closest idiomatic type mapping to a JSON Object type for .NET is a Dictionary, so we use that here.</remarks>
+    public async Task<bool> check(string path, Dictionary<string, object> input)
     {
-        return queryMachinery<T>(Input.CreateMapOfany(input), path);
+        return await evaluate<bool>(path, input);
     }
 
     /// <summary>
-    /// Evaluate a rule, using the provided input string, then coerce the result to type T.
+    /// Evaluate a rule, then coerce the result to type T.
     /// </summary>
-    /// <param name="input">The input string OPA will use for evaluating the rule.</param>
     /// <param name="path">The rule to evaluate. (Example: "app/rbac")</param>
-    /// <returns>Result, as a boolean, or nil in the case of a query failure.</returns>
-    public T? query<T>(string input, string path)
+    /// <returns>Result, as an instance of T, or null in the case of a query failure.</returns>
+    public async Task<T?> evaluate<T>(string path)
     {
-        return queryMachinery<T>(Input.CreateStr(input), path);
+        return await queryMachinery<T>(path, Input.CreateNull());
```

Comment:
💭 Is this Null a JSON `null` or the C#-equivalent of Go's `nil`?

Reply:
More a JSON `null`, I think? :thinking: 

Both the `OpenApi.Result` and `OpenApi.Input` types are generated as [Tagged Unions](https://en.wikipedia.org/wiki/Tagged_union).

The `Input.CreateNull()` constructor basically leaves all of the `Input` object's value fields null, while setting the tag field to explicitly indicate that JSON-level null value is being passed through.